recipe/migrations/numpy2.yaml

```diff
@@ -0,0 +1,78 @@
+__migrator:
+  build_number: 1
+  kind: version
+  commit_message: |
+    Rebuild for numpy 2.0
+    
+    TL;DR: The way we build against numpy has changed as of numpy 2.0. This bot
+    PR has updated the recipe to account for the changes (see below for details).
+    The numpy 2.0 package itself is currently only available from a special release
+    channel (`conda-forge/label/numpy_rc`) and will not be available on the main
+    `conda-forge` channel until the release of numpy 2.0 GA.
+    
+    The biggest change is that we no longer need to use the oldest available numpy
+    version at build time in order to support old numpy version at runtime - numpy
+    will by default use a compatible ABI for the oldest still-supported numpy versions.
+    
+    Additionally, we no longer need to use `{{ pin_compatible("numpy") }}` as a
+    run requirement - this has been handled for more than two years now by a
+    run-export on the numpy package itself. The migrator will therefore remove
+    any occurrences of this.
+    
+    However, by default, building against numpy 2.0 will assume that the package
+    is compatible with numpy 2.0, which is not necessarily the case. You should
+    check that the upstream package explicitly supports numpy 2.0, otherwise you
+    need to add a `- numpy <2` run requirement until that happens (check numpy
+    issue 26191 for an overview of the most important packages).
+    
+    Note that the numpy release candidate promises to be ABI-compatible with the
+    final 2.0 release. This means that building against 2.0.0rc1 produces packages
+    that can be published to our main channels.
+    
+    If you already want to use the numpy 2.0 release candidate yourself, you can do
+    ```
+    conda config --add channels conda-forge/label/numpy_rc
+    ```
+    or add this channel to your `.condarc` file directly.
+    
+    To-Dos:
+      * [ ] Match run-requirements for numpy (i.e. check upstream `pyproject.toml` or however the project specifies numpy compatibility)
+        * If upstream is not yet compatible with numpy 2.0, add `numpy <2` upper bound under `run:`.
+        * If upstream is already compatible with numpy 2.0, nothing else should be necessary in most cases.
+        * If upstream requires a minimum numpy version newer than 1.19, you can add `numpy >=x.y` under `run:`.
```

Comment:
I'm not sure how much we want to talk about lower bounds, because this is often not necessarily correct upstream either. But I think we need to mention it